File: algorithms/223.py
# https://leetcode.com/problems/rectangle-area


class Solution(object):
    def computeArea(self, A, B, C, D, E, F, G, H):
        """
        :type A: int
        :type B: int
        :type C: int
        :type D: int
        :type E: int
        :type F: int
        :type G: int
        :type H: int
        :rtype: int
        """

        r_abcd = (C - A) * (D - B)
        r_efgh = (G - E) * (H - F)
        r_overlapped = 0
        if C > E and G > A and H > B and D > F:
            r_overlapped = min(C - E, G - A, C - A, G - E) * min(H - B, D - F, D - B, H - F)

        return r_abcd + r_efgh - r_overlapped

File: algorithms/223.swift
// https://leetcode.com/problems/rectangle-area

class Solution {
    func computeArea(A: Int, _ B: Int, _ C: Int, _ D: Int, _ E: Int, _ F: Int, _ G: Int, _ H: Int) -> Int {
        let r_abcd = (C-A) * (D-B)
        let r_efgh = (G-E) * (H-F)
        var r_overlapped = 0
        if C>E && G>A && H>B && D>F { r_overlapped = min(C-E, G-A, C-A, G-E) * min(H-B, D-F, D-B, H-F) }
        return r_abcd + r_efgh - r_overlapped
    }
}
